### PR TITLE
bluetooth: scan: Add connect_if_match flag update function.

### PR DESCRIPTION
**Pull request by maje-emb**
Commit adds connet_if_match flag update function.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -466,6 +466,12 @@ Bluetooth libraries and services
 
     * The :ref:`bt_mesh_le_pair_resp_readme` model to allow passing a passkey used in LE pairing over a mesh network.
 
+ :ref:`nrf_bt_scan_readme` library:
+
+  * Added:
+
+  * The :c:func:`bt_scan_update_connect_if_match` function to update the autoconnect flag after a filter match.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:473]:
```suggestion
  * Added the :c:func:`bt_scan_update_connect_if_match` function to update the autoconnect flag after a filter match.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by dchat-nordic — APPROVED** [review]

**Review by KAGA164 — CHANGES_REQUESTED** [review]

File: include/bluetooth/scan.h [hunk 2]
```diff
@@ -562,6 +562,20 @@ int bt_scan_blocklist_device_add(const bt_addr_le_t *addr);
  */
 void bt_scan_blocklist_clear(void);
 
+
+#if CONFIG_BT_CENTRAL
```
[the remark this answers is not in the document]
Comment (maje-emb) [reply include/bluetooth/scan.h:566]: This function is useless e.g when BT_OBSERVER is enabled. Therefore there is conditional compilation, as for CONFIG_BT_SCAN_FILTER_ENABLE in this file.

[the remark this answers is not in the document]
Comment (KAGA164) [reply include/bluetooth/scan.h:566]: Still it can be under conditional compilation in source file

**Review by grochu — APPROVED** [review]
Looks good - if it simplifies the implementation in some use cases why not add it, the cost is low.

File: include/bluetooth/scan.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (maje-emb) [reply include/bluetooth/scan.h]: Yes, this is not needed in the header file.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -499,6 +499,10 @@ Bluetooth libraries and services
 
     * The :ref:`bt_mesh_le_pair_resp_readme` model to allow passing a passkey used in LE pairing over a mesh network.
 
+ :ref:`nrf_bt_scan_readme` library:
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:502]:
```suggestion
 :ref:`nrf_bt_scan_readme`:
```
Not so sure about this. This library is called Scanning module. Feels a bit awkward to call it Scanning module library.

**Review by peknis — APPROVED** [review]

**Pull request merged**